NotesApplication/Screens/Home/NotesViewController.swift

```diff
@@ -0,0 +1,189 @@
+    }
+    
+    func setNavigationTitle() {
+        navigationItem.title = "Notes"
+        navigationController?.navigationBar.prefersLargeTitles = true
+        navigationController?.navigationBar.largeTitleTextAttributes = [.foregroundColor: UIColor.white]
+        navigationController?.navigationBar.setBackgroundImage(UIImage(), for: .default)
+    }
+    
+    // MARK: - Custom Alert
+    private func showDeleteAlert(for indexPath: IndexPath) {
+        let alertView = CustomAlertView()
+        alertView.delegate = self
+        alertView.frame = view.bounds
+        alertView.tag = indexPath.row
+        view.addSubview(alertView)
+    }
+    
+    // MARK: - CustomAlertViewDelegate Methods
+    func didTapDeleteButton(in cell: NoteTableViewCell) {
+        guard let indexPath = tableView.indexPath(for: cell) else { return }
+        showDeleteAlert(for: indexPath)
+    }
+    
+    func didTapDeleteButton() {
+        guard let alertView = view.subviews.first(where: { $0 is CustomAlertView }),
+              let index = alertView.tag as Int? else { return }
+        
+        viewModel.deleteNote(at: index)
+        tableView.reloadData()
+        alertView.removeFromSuperview()
+    }
+    
+    func didTapCancelButton() {
+        guard let alertView = view.subviews.first(where: { $0 is CustomAlertView }) else { return }
+        alertView.removeFromSuperview()
+    }
+}
+
+// MARK: - NotesViewModelDelegate
+extension NotesViewController: NotesViewModelDelegate {
+    func notesDidUpdate() {
+        DispatchQueue.main.async { [weak self] in
+            self?.tableView.reloadData()
+        }
+    }
+}
+
+// MARK: - TableView DataSource
+extension NotesViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return viewModel.numberOfNotes()
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: NoteTableViewCell.identifier, for: indexPath) as? NoteTableViewCell else {
+            return UITableViewCell()
+        }
+        
+        let note = viewModel.note(at: indexPath.row)
+        cell.configure(with: note)
+        cell.delegate = self
+        cell.setupDeleteButton()
+        
+        cell.selectionStyle = .none
+        cell.backgroundColor = .clear
+        
+        return cell
+    }
+}
+// MARK: - UITableViewDelegate
+extension NotesViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let note = viewModel.note(at: indexPath.row)
+        let editorVC = NotesEditorViewController(note: note, index: indexPath.row)
+        editorVC.delegate = self
+        navigationController?.pushViewController(editorVC, animated: true)
+    }
+}
+
+// MARK: - Note Editor Delegate
```

Comment:
//MARK: - NotesEditorDelegate

mark title should be name of the delegate as you have this // MARK: - UITableViewDelegate